### PR TITLE
#54602 - View producer summary by compliance year

### PR DESCRIPTION
**Pull request by SFWMuhammadI**

### CODE REVIEW

File: src/EA.Weee.RequestHandlers.Tests.DataAccess/Admin/GetProducerDetails/GetProducerComplianceYearDataAccessTests.cs [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+                // Assert
+                Assert.Equal(2, result.Count);
+                Assert.Contains(2016, result);
+                Assert.Contains(2017, result);
+            }
+        }
+
+        [Fact]
+        public async Task GetComplianceYears_ProducerRegisteredWithTwoSchemesInSameYear_ReturnsOneInstanceOfComplianceYear()
```
Comment (SFWChrisSmith) [inline src/EA.Weee.RequestHandlers.Tests.DataAccess/Admin/GetProducerDetails/GetProducerComplianceYearDataAccessTests.cs]: Good thought :)

File: src/EA.Weee.RequestHandlers.Tests.DataAccess/Admin/GetProducerDetails/GetProducerDetailsHandlerTests.cs [hunk 2]
```diff
@@ -1,62 +0,0 @@
...
-
-        public GetProducerDetailsHandlerTests()
-        {
-            authorization = A.Fake<IWeeeAuthorization>();
-            mapper = A.Fake<IMapper>();
-        }
-
-        [Theory(Skip = "Should only be run manually to check the performance of this handler with a populated database")]
-        [InlineData(3000, "WEE/MM0004AA")]
-        public async void PerformanceTest(int expectedUnderMilliseconds, string producerRegistrationNumber)
```
Comment (SFWChrisSmith) [inline src/EA.Weee.RequestHandlers.Tests.DataAccess/Admin/GetProducerDetails/GetProducerDetailsHandlerTests.cs]: Any reason that this was removed?

Comment (SFWMuhammadI) [inline src/EA.Weee.RequestHandlers.Tests.DataAccess/Admin/GetProducerDetails/GetProducerDetailsHandlerTests.cs]: I have now re-added the test as discussed.

File: src/EA.Weee.RequestHandlers.Tests.DataAccess/Admin/GetProducerDetails/GetProducerComplianceYearDataAccessTests.cs (hunk 1 above)
Comment (SFWMuhammadI) [inline src/EA.Weee.RequestHandlers.Tests.DataAccess/Admin/GetProducerDetails/GetProducerComplianceYearDataAccessTests.cs]: :)

File: src/EA.Weee.RequestHandlers.Tests.DataAccess/Admin/GetProducerDetails/GetProducerComplianceYearDataAccessTests.cs [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+                var result = await dataAccess.GetComplianceYears("AAA");
+
+                // Assert
+                Assert.Single(result);
+                Assert.Contains(2016, result);
+            }
+        }
+
+        [Fact]
+        public async Task GetComplianceYears_ReturnsComplianceYearForSpecifiedProducerRegistrationNumberOnly()
```
Comment (SFWMuhammadI) [inline src/EA.Weee.RequestHandlers.Tests.DataAccess/Admin/GetProducerDetails/GetProducerComplianceYearDataAccessTests.cs]: As discussed, this is as expected and will return all registered producers, regardless whether they have any data returns.

**Pull request merged**